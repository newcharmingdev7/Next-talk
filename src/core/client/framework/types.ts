// TODO: (@cvle) Extract useful common types into its own package.
export { Omit, Overwrite, PropTypesOf } from "coral-ui/types";
export { DeepPartial } from "coral-common/types";

export type RelayEnumLiteral<T> = keyof T | "%future added value";
