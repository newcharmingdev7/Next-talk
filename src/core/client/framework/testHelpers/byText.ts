import React from "react";
import { ReactTestInstance } from "react-test-renderer";

import matchText, { TextMatchOptions, TextMatchPattern } from "./matchText";

const matcher = (pattern: TextMatchPattern, options?: TextMatchOptions) => (
  i: ReactTestInstance
) => {
  // Only look at dom components.
  if (typeof i.type !== "string") {
    return false;
  }
  if (!i.props.children) {
    return false;
  }
  const children = React.Children.toArray(i.props.children);
  for (const c of children) {
    if (typeof c === "string" && matchText(pattern, c, options)) {
      return true;
    }
  }
  return false;
};

export function getByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  return container.find(matcher(pattern, options));
}

export function getAllByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  const results = container.findAll(matcher(pattern, options));
  if (!results.length) {
    throw new Error(`Couldn't find text ${pattern}`);
  }
  return results;
}

export function queryByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  const results = container.findAll(matcher(pattern, options));
  if (!results.length) {
    return null;
  }
  return results[0];
}

export function queryAllByText(
  container: ReactTestInstance,
  pattern: TextMatchPattern,
  options?: TextMatchOptions
) {
  return container.findAll(matcher(pattern, options));
}
