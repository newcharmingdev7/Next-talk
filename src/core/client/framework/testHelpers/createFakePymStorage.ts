import { PymStorage } from "talk-framework/lib/storage";

export class FakeStorage implements PymStorage {
  public store: Record<string, string> = {};

  public setItem(key: string, value: string) {
    this.store[key] = value;
    return Promise.resolve();
  }
  public removeItem(key: string) {
    delete this.store[key];
    return Promise.resolve();
  }
  public getItem(key: string) {
    return Promise.resolve(this.store[key] || null);
  }
}

export default function createFakePymStorage() {
  return new FakeStorage();
}
