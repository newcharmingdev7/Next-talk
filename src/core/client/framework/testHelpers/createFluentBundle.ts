import { FluentBundle } from "fluent/compat";
import fs from "fs";
import path from "path";

// These locale prefixes are always loaded.
const commonPrefixes = ["common", "framework"];

function decorateErrorWhenMissing(bundle: FluentBundle) {
  const originalHasMessage = bundle.hasMessage;
  const originalGetMessage = bundle.getMessage;
  const missing: string[] = [];
  bundle.hasMessage = (id: string) => {
    const result = originalHasMessage.apply(bundle, [id]);
    if (!result) {
      const msg = `${bundle.locales} translation for key "${id}" not found`;
      // tslint:disable-next-line:no-console
      console.error(msg);
      missing.push(id);
    }
    // Even if it is missing, we say it is available and later return a descriptive error
    // string as the translation.
    return true;
  };
  bundle.getMessage = (id: string) => {
    if (missing.indexOf(id) !== -1) {
      return `Missing translation "${id}"`;
    }
    return originalGetMessage.apply(bundle, [id]);
  };
  return bundle;
}

function createFluentBundle(
  target: string,
  pathToLocale: string
): FluentBundle {
  const bundle = new FluentBundle("en-US");
  const files = fs.readdirSync(pathToLocale);
  const prefixes = commonPrefixes.concat(target);
  files.forEach(f => {
    prefixes.forEach(prefix => {
      if (f.startsWith(prefix)) {
        bundle.addMessages(require(path.resolve(pathToLocale, f)));
      }
    });
  });
  return decorateErrorWhenMissing(bundle);
}

export default createFluentBundle;
