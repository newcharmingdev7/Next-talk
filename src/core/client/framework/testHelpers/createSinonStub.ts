import sinon, { SinonStub } from "sinon";

export default function createSinonStub(
  ...callbacks: Array<(s: SinonStub) => void>
): SinonStub {
  const stub = sinon.stub();
  callbacks.forEach(cb => cb(stub));
  return stub;
}
