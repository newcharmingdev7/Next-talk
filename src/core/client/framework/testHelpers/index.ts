export {
  default as createRelayEnvironment,
  CreateRelayEnvironmentParams,
} from "./createRelayEnvironment";
export { default as createFluentBundle } from "./createFluentBundle";
export {
  default as removeFragmentRefs,
  NoFragmentRefs,
} from "./removeFragmentRefs";
