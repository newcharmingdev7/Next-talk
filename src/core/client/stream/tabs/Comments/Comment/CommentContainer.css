$commenterActionEditColor: var(--palette-primary-500);
$commenterActionEditColorActive: var(--palette-primary-300);

.root {
  &:focus {
    outline: none;
  }
}
.tagIcon {
  margin-right: var(--spacing-1);
}

.answeredTag {
  color: #03ab61;
  background: #e6faf1;

  border-color: #03ab61;
  border-style: solid;
  border-radius: 2px;
  border-width: 1px;

  text-transform: uppercase;
}

.editButton {
  color: $commenterActionEditColor;
  &:hover,
  &.mouseHover {
    color: $commenterActionEditColorActive;
  }
  &:active,
  &.active {
    color: $commenterActionEditColorActive;
  }
}

.editIcon {
  margin-right: var(--spacing-1);
}

.actionBar {
  min-width: 100px;
  @media (min-width: $breakpoints-xs) {
    min-width: 150px;
  }
}

.actionButton {
  padding: var(--spacing-1);
}

.staticUsername {
  color: var(--palette-text-100);

  margin-right: var(--spacing-2);
}

.avatarContainer {
  width: 24px;
  height: 24px;
}

.avatar {
  width: 100%;
  height: 100%;
  border-radius: 4px;
}

.usernameButton {
  overflow: hidden;
  text-overflow: ellipsis;
  text-align: left;
  max-width: 100%;
}

.traversalFocus .commentSeenEnabled {
  background-color: $colors-yellow-500;
}

.commentSeenEnabled {
  padding-top: 8px;
  padding-bottom: 6px;
  transition: background 600ms ease-out;
}

.notSeen {
  background-color: $colors-yellow-100;
}
