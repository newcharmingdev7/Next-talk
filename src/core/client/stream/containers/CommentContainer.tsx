import React, { Component } from "react";
import { graphql } from "react-relay";

import withFragmentContainer from "talk-framework/lib/relay/withFragmentContainer";
import { PropTypesOf } from "talk-framework/types";
import { CommentContainer_asset as AssetData } from "talk-stream/__generated__/CommentContainer_asset.graphql";
import { CommentContainer_comment as CommentData } from "talk-stream/__generated__/CommentContainer_comment.graphql";

import Comment from "../components/Comment";
import ReplyButton from "../components/Comment/ReplyButton";
import ReplyCommentFormContainer from ".//ReplyCommentFormContainer";
import PermalinkButtonContainer from "./PermalinkButtonContainer";

interface InnerProps {
  comment: CommentData;
  asset: AssetData;
}

interface State {
  showReplyDialog: boolean;
}

export class CommentContainer extends Component<InnerProps, State> {
  public state = {
    showReplyDialog: false,
  };

  private openReplyDialog = () => {
    this.setState(state => ({
      showReplyDialog: true,
    }));
  };

  private closeReplyDialog = () => {
    this.setState(state => ({
      showReplyDialog: false,
    }));
  };

  public render() {
    const { comment, asset, ...rest } = this.props;
    const { showReplyDialog } = this.state;
    return (
      <>
        <Comment
          {...rest}
          {...comment}
          footer={
            <>
              <ReplyButton
                onClick={this.openReplyDialog}
                active={showReplyDialog}
              />
              <PermalinkButtonContainer commentID={comment.id} />
            </>
          }
        />
        {showReplyDialog && (
          <ReplyCommentFormContainer
            comment={comment}
            asset={asset}
            onClose={this.closeReplyDialog}
          />
        )}
      </>
    );
  }
}

const enhanced = withFragmentContainer<InnerProps>({
  asset: graphql`
    fragment CommentContainer_asset on Asset {
      ...ReplyCommentFormContainer_asset
    }
  `,
  comment: graphql`
    fragment CommentContainer_comment on Comment {
      id
      author {
        username
      }
      body
      createdAt
      ...ReplyCommentFormContainer_comment
    }
  `,
})(CommentContainer);

export type CommentContainerProps = PropTypesOf<typeof enhanced>;
export default enhanced;
