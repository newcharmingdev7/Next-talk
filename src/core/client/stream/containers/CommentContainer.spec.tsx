import { shallow } from "enzyme";
import React from "react";

import { removeFragmentRefs } from "talk-framework/testHelpers";
import { PropTypesOf } from "talk-framework/types";

import { CommentContainer } from "./CommentContainer";

// Remove relay refs so we can stub the props.
const CommentContainerN = removeFragmentRefs(CommentContainer);

it("renders username and body", () => {
  const props: PropTypesOf<typeof CommentContainerN> = {
    asset: {
      id: "asset-id",
    },
    comment: {
      id: "comment-id",
      author: {
        username: "Marvin",
      },
      body: "Woof",
      createdAt: "1995-12-17T03:24:00.000Z",
    },
    indentLevel: 1,
  };

  const wrapper = shallow(<CommentContainerN {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders body only", () => {
  const props: PropTypesOf<typeof CommentContainerN> = {
    asset: {
      id: "asset-id",
    },
    comment: {
      id: "comment-id",
      author: {
        username: null,
      },
      body: "Woof",
      createdAt: "1995-12-17T03:24:00.000Z",
    },
    indentLevel: 1,
  };

  const wrapper = shallow(<CommentContainerN {...props} />);
  expect(wrapper).toMatchSnapshot();
});
