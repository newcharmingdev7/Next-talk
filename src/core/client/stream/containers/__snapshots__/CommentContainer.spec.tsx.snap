// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders body only 1`] = `
<React.Fragment>
  <IndentedComment
    author={
      Object {
        "username": null,
      }
    }
    body="Woof"
    createdAt="1995-12-17T03:24:00.000Z"
    footer={
      <React.Fragment>
        <ReplyButton
          active={false}
          id="comments-commentContainer-replyButton-comment-id"
          onClick={[Function]}
        />
        <withContext(withLocalStateContainer(PermalinkContainer))
          commentID="comment-id"
        />
      </React.Fragment>
    }
    indentLevel={1}
  />
</React.Fragment>
`;

exports[`renders username and body 1`] = `
<React.Fragment>
  <IndentedComment
    author={
      Object {
        "username": "Marvin",
      }
    }
    body="Woof"
    createdAt="1995-12-17T03:24:00.000Z"
    footer={
      <React.Fragment>
        <ReplyButton
          active={false}
          id="comments-commentContainer-replyButton-comment-id"
          onClick={[Function]}
        />
        <withContext(withLocalStateContainer(PermalinkContainer))
          commentID="comment-id"
        />
      </React.Fragment>
    }
    indentLevel={1}
  />
</React.Fragment>
`;
